pkg/maps/nat/per_cluster_nat.go

```diff
@@ -0,0 +1,413 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package nat
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+const (
+	mapNamePerClusterSnat4Global = "cilium_per_cluster_snat_v4_external"
+	mapNamePerClusterSnat6Global = "cilium_per_cluster_snat_v6_external"
+	innerMapNamePrefix4          = MapNameSnat4Global + "_"
+	innerMapNamePrefix6          = MapNameSnat6Global + "_"
+	perClusterNATMapMaxEntries   = cmtypes.ClusterIDMax + 1
+)
+
+// Global interface to interact with IPv4 and v6 NAT maps. We can choose the
+// implementation of this at startup time by choosing InitPerClusterNATMaps
+// or InitDummyPerClusterNATMaps for initialization.
+var PerClusterNATMaps PerClusterNATMapsIface
+
+// An interface to interact with the global map.
+type PerClusterNATMapsIface interface {
```

Comment:
I do not typically see Go interfaces called "Iface". Any better naming for this?

Reply:
Yep, Timo pointed out the same in this comment https://github.com/cilium/cilium/pull/22857#discussion_r1057240088. I'll rename it to `PerClusterNATMapper`